test/integration/guest_can_change_quanity_in_cart_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class GuestCanChangeQuanityInCartTest < ActionDispatch::IntegrationTest
+  test "guest visits cart and adds more items" do
+    create_and_add_item_to_cart
+    visit cart_path
+    fill_in "Quantity", with: "4"
+    click_button "Save changes"
+    # save_and_open_page
+    assert_equal '/cart', current_path
+    assert page.has_content? "Total: $#{@item.price * 4}"
+    refute page.has_content? "Total: $#{@item.price * 1}"
+
+    fill_in "Quantity", with: "2"
+    click_button "Save changes"
+
+    assert_equal '/cart', current_path
+    assert page.has_content? "Total: $#{@item.price * 2}"
+    refute page.has_content? "Total: $#{@item.price * 4}"
+  end
+
+  def create_and_add_item_to_cart
+    @item = create(:item)
+
+    visit item_path(@item)
+    click_button "Add to Cart"
+  end
+  #
+  # Background: My cart has an item in it
+  #     As a visitor
+  #     When I visit "/cart"
+  #     Then I should see my item with a quantity of 1
+  #     And when I increase the quantity
+  #     Then my current page should be '/cart'
+  #     And that item's quantity should reflect the increase
+  #     And the subtotal for that item should increase
```

Comment:
I just realized that we are not showing subtotals, so we need to add that to our cart page (within the item partial I assume).